### PR TITLE
[#7702] improvement(core) : Update strToSeq to return empty lists for empty strings

### PR DESCRIPTION
**Pull request by georgereuben**
### What changes were proposed in this pull request?
Update strToSeq to return empty lists for empty strings

### Why are the changes needed?
Fix: #7702 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Added a unit test

### CODE REVIEW

**Comment by yuqi1129** [comment]
@georgereuben 
Is this PR ready for review?

File: core/src/test/java/org/apache/gravitino/config/TestConfigEntry.java [hunk 1]
```diff
@@ -168,4 +169,11 @@ public void testCheckValue() {
     Assertions.assertThrows(
         IllegalArgumentException.class, () -> testConfOptional.readFrom(configMap));
   }
+
+  @Test
+  public void testStrToSeqEmptyString() {
+    ConfigEntry<String> conf = new ConfigBuilder("gravitino.test.seq").stringConf().create();
+    List<String> result = conf.strToSeq("", s -> s);
```
Comment (yuqi1129) [inline core/src/test/java/org/apache/gravitino/config/TestConfigEntry.java:176]: You may need to test a `null` value.

**Comment by georgereuben** [comment]
> @georgereuben Is this PR ready for review?

Yes, I was facing issues with the `testCheckValue` test, so I have made some changes to make it work with the new strToSeq behavior.

The original test used a list containing only an empty string [""], which serialized to an empty string "". With the new behavior, `strToSeq("") ` returns an empty list [] instead of [""], so validation never runs.

Changed the test to use ["valid", "", "another"] which serializes to "valid,,another". This still tests validation on empty elements while  working with the updated strToSeq behavior.

I have updated the test in the [latest commit](https://github.com/apache/gravitino/pull/7755/commits/bd3d3f1b8ea734513fe8d4c08bd3bb89d98b5d49) . Please let me know if this change looks good

File: core/src/test/java/org/apache/gravitino/config/TestConfigEntry.java (hunk 1 above)
Comment (georgereuben) [reply core/src/test/java/org/apache/gravitino/config/TestConfigEntry.java:176]: Added a test to check that null strings return empty lists as well

**Review by yuqi1129 — APPROVED** [review]
LGTM
